app/scripts/modules/core/src/pipeline/executions/execution/Execution.tsx

```diff
@@ -471,29 +475,37 @@ export class Execution extends React.Component<IExecutionProps, IExecutionState>
             )}
           </div>
 
-          <div className="execution-parameters-button">
-            <a className="clickable" onClick={this.toggleParams}>
-              <span
-                className={`small glyphicon ${
-                  parametersAndArtifactsExpanded ? 'glyphicon-chevron-down' : 'glyphicon-chevron-right'
-                }`}
+          {showParamsAndArtifacts && (
+            <>
+              <div className="execution-parameters-button">
+                <a className="clickable" onClick={this.toggleParams}>
+                  <span
+                    className={`small glyphicon ${
+                      parametersAndArtifactsExpanded ? 'glyphicon-chevron-down' : 'glyphicon-chevron-right'
+                    }`}
+                  />
+                  {parametersAndArtifactsExpanded ? '' : 'View All '}
+                  {showParameters &&
+                    showArtifacts &&
+                    `Parameters/Artifacts (${displayableParameters.length}/${resolvedExpectedArtifacts.length})`}
+                  {showParameters && !showArtifacts && `Parameters (${displayableParameters.length})`}
+                  {!showParameters && showArtifacts && `Artifacts (${resolvedExpectedArtifacts.length})`}
```

Comment:
tiny nit: this set of conditional renders is kind of messy/tough to read and i almost want it to be a function that can return each of these in a nice control flow instead of this grab bag of JSX.

Reply:
This render method was getting a little out of control, just kidding, it was out of control a long time ago. Moved into a separate component.

Reply:
![clean](https://media.giphy.com/media/v5V3XxJCV0pFu/giphy.gif)